File: app/src/main/java/com/letit0or1/akimaleo/eyedoctor/colorblind/entity/DataCollection.java
package com.letit0or1.akimaleo.eyedoctor.colorblind.entity;

import android.content.Context;
import android.util.Log;

import com.letit0or1.akimaleo.eyedoctor.R;

import java.lang.reflect.Field;
import java.util.ArrayList;
import java.util.Collections;

//Class using for accessing data in application storage and structuring it not in
//view controllers
public class DataCollection {
    //singleton class
    private static DataCollection ourInstance = new DataCollection();
    private Context context;

    public static DataCollection getInstance() {
        return ourInstance;
    }

    public void setContext(Context context) {
        this.context = context;
    }

    private DataCollection() {
    }

    public ArrayList<DataItem> getData() {
//get all fields of drawable pat
        final R.drawable drawableResources = new R.drawable();
        final Class<R.drawable> c = R.drawable.class;
        final Field[] fields = c.getDeclaredFields();

        ArrayList<DataItem> data = new ArrayList();
        //get Strings data from a String resources
        String[] desk = context.getResources().getStringArray(R.array.diagnostic_data);

        //collecting data ind ArrayList
        for (int i = 0, max = fields.length; i < max; i++) {
            final int resourceId;
            try {
                if (fields[i].getName().startsWith("plate")) {
                    resourceId = fields[i].getInt(drawableResources);
                    data.add(new DataItem(null, resourceId));

                } else continue;
            } catch (Exception e) {
                continue;
            }
        /* make use of resourceId for accessing Drawables here */
        }
//        sorting collection and adding description reference for image
        Collections.sort(data);
        for (int i = 0; i < data.size(); i++) {
            data.get(i).setDescription(desk[i]);
            int plateNumberObject = Integer.parseInt(DataCollection.getInstance().getContext().getResources().getResourceEntryName((data.get(i)).getImageResource()).replace("plate", "").intern());
            Log.i("num", "" + plateNumberObject);
            switch (i) {
                case 0:
                    data.get(i).setAnswer(new int[]{12, 0, 0});
                    break;
                case 1:
                    data.get(i).setAnswer(new int[]{8, 3, 0});
                    break;
                case 2:
                    data.get(i).setAnswer(new int[]{29, 70, 0});
                    break;
                case 3:
                    data.get(i).setAnswer(new int[]{5, 2, 0});
                    break;
                case 4:
                    data.get(i).setAnswer(new int[]{3, 5, 0});
                    break;
                case 5:
                    data.get(i).setAnswer(new int[]{15, 17, 0});
                    break;
                case 6:
                    data.get(i).setAnswer(new int[]{74, 21, 0});
                    break;
                case 7:
                    data.get(i).setAnswer(new int[]{6, 0, 0});
                    break;
                case 8:
                    data.get(i).setAnswer(new int[]{45, 0, 0});
                    break;
                case 9:
                    data.get(i).setAnswer(new int[]{5, 0, 0});
                    break;
                case 10:
                    data.get(i).setAnswer(new int[]{7, 0, 0});
                    break;
                case 11:
                    data.get(i).setAnswer(new int[]{16, 0, 0});
                    break;
                case 12:
                    data.get(i).setAnswer(new int[]{73, 0, 0});
                    break;
                case 13:
                    data.get(i).setAnswer(new int[]{0, 5, 0});
                    break;
                case 14:
                    data.get(i).setAnswer(new int[]{0, 45, 0});
                    break;
                case 15:
                    data.get(i).setAnswer(new int[]{26, 2, 0});
                    break;
                case 16:
                    data.get(i).setAnswer(new int[]{42, 2, 0});
                    break;
            }
        }

        return data;
    }

    public Context getContext() {
        return context;
    }

}

File: app/src/main/java/com/letit0or1/akimaleo/eyedoctor/colorblind/entity/DataCollection.kt
package com.letit0or1.akimaleo.eyedoctor.colorblind.entity

import android.content.Context
import android.util.Log
import com.letit0or1.akimaleo.eyedoctor.R
import java.util.*

//Class using for accessing data in application storage and structuring it not in
//view controllers
class DataCollection {
    companion object {
        //singleton class
        val instance = DataCollection()
    }

    var context: Context? = null

    //get all fields of drawable pat
    //get Strings data from a String resources
    //collecting data ind ArrayList
    /* make use of resourceId for accessing Drawables here *///        sorting collection and adding description reference for image
    val data: ArrayList<DataItem>
        get() {
            val drawableResources = R.drawable()
            val c = R.drawable::class.java
            val fields = c.declaredFields

            val data = ArrayList<DataItem>()
            val desk = context!!.resources.getStringArray(R.array.diagnostic_data)

            for (i in 0..fields.size) {
                val resourceId: Int
                try {
                    if (fields[i].name.startsWith("plate")) {
                        resourceId = fields[i].getInt(drawableResources)
                        data.add(DataItem(null, resourceId))
                    } else
                        continue
                } catch (e: Exception) {
                    continue
                }
            }

            Collections.sort<DataItem>(data)
            for (i in data.indices) {
                data.get(i).setDescription(desk[i])
                val plateNumberObject = Integer.parseInt(DataCollection.instance.context!!.getResources().getResourceEntryName(data.get(i).getImageResource()).replace("plate", "").intern())
                Log.i("num", "" + plateNumberObject)
                when (i) {
                    0 -> data.get(i).setAnswer(intArrayOf(12, 0, 0))
                    1 -> data.get(i).setAnswer(intArrayOf(8, 3, 0))
                    2 -> data.get(i).setAnswer(intArrayOf(29, 70, 0))
                    3 -> data.get(i).setAnswer(intArrayOf(5, 2, 0))
                    4 -> data.get(i).setAnswer(intArrayOf(3, 5, 0))
                    5 -> data.get(i).setAnswer(intArrayOf(15, 17, 0))
                    6 -> data.get(i).setAnswer(intArrayOf(74, 21, 0))
                    7 -> data.get(i).setAnswer(intArrayOf(6, 0, 0))
                    8 -> data.get(i).setAnswer(intArrayOf(45, 0, 0))
                    9 -> data.get(i).setAnswer(intArrayOf(5, 0, 0))
                    10 -> data.get(i).setAnswer(intArrayOf(7, 0, 0))
                    11 -> data.get(i).setAnswer(intArrayOf(16, 0, 0))
                    12 -> data.get(i).setAnswer(intArrayOf(73, 0, 0))
                    13 -> data.get(i).setAnswer(intArrayOf(0, 5, 0))
                    14 -> data.get(i).setAnswer(intArrayOf(0, 45, 0))
                    15 -> data.get(i).setAnswer(intArrayOf(26, 2, 0))
                    16 -> data.get(i).setAnswer(intArrayOf(42, 2, 0))
                }
            }

            return data
        }


}
